src/VecSim/spaces/IP_space.cpp

```diff
@@ -30,36 +30,31 @@ dist_func_t<float> IP_FP32_GetDistFunc(size_t dim, const Arch_Optimization arch_
         return ret_dist_func;
     }
 #ifdef CPU_FEATURES_ARCH_X86_64
-
-    switch (arch_opt) {
-    case ARCH_OPT_AVX512_BW_VBMI2:
-    case ARCH_OPT_AVX512_F:
+    auto features = arch_opt.features;
```

Comment:
if we return in every if block, the pre declaration of `ret_dist_func` is redundant